build-system/tasks/visual-diff/index.js

```diff
@@ -469,59 +471,59 @@ async function snapshotWebpages(browser, webpages) {
+            const responseWatcher = new Promise((resolve, reject) => {
+              const responseTimeout = setTimeout(() => {
+                reject(
+                  new puppeteer.TimeoutError(
+                    `Response was not received in test ${testName} for page ` +
+                      `${webpage.url} after ${NAVIGATE_TIMEOUT_MS}ms`
+                  )
+                );
+              }, NAVIGATE_TIMEOUT_MS);
+
+              page.once('response', (response) => {
+                log(
+                  'verbose',
+                  'Response for url',
+                  colors.yellow(response.url()),
+                  'with status',
+                  colors.cyan(response.status()),
+                  colors.cyan(response.statusText())
+                );
+                clearTimeout(responseTimeout);
+                resolve();
+              });
+            });
+
+            log('verbose', 'Navigating to page', colors.yellow(webpage.url));
+            await Promise.all([
+              responseWatcher,
+              page.goto(fullUrl, {waitUntil: 'networkidle2'}),
+            ]);
 
-          page.once('response', async (response) => {
             log(
               'verbose',
-              'Response for url',
-              colors.yellow(response.url()),
-              'with status',
-              colors.cyan(response.status()),
-              colors.cyan(response.statusText())
+              'Page navigation of test',
+              colors.yellow(name),
+              'is done, verifying page'
             );
-            clearTimeout(responseTimeout);
-            resolve();
-          });
-        });
-
-        log('verbose', 'Navigating to page', colors.yellow(webpage.url));
-        await Promise.all([
-          responseWatcher,
-          page.goto(fullUrl, {waitUntil: 'networkidle2'}),
-        ]);
-      })()
-        .then(() => {
-          log(
-            'verbose',
-            'Page navigation of test',
-            colors.yellow(name),
-            'is done, verifying page'
-          );
-        })
-        .catch((navigationError) => {
-          hasWarnings = true;
-          addTestError(
-            testErrors,
-            name,
-            'The browser test runner failed to complete the navigation ' +
-              'to the test page',
-            navigationError,
-            consoleMessages
-          );
-          if (!isTravisBuild()) {
-            log('warning', 'Continuing to verify page regardless...');
+          } catch (navigationError) {
+            hasWarnings = true;
+            addTestError(
+              testErrors,
+              name,
+              'The browser test runner failed to complete the navigation ' +
+                'to the test page',
+              navigationError,
+              consoleMessages
+            );
+            if (!isTravisBuild()) {
+              log('warning', 'Continuing to verify page regardless...');
+            }
```

Comment:
Condition can be dropped, just print it regardless
(this is a relic from when visual diffs were running as part of a shared job with other tests and we didn't want to "spam" the logs)